scientific/src/commonTest/kotlin/ScientificValueTest.kt

```diff
@@ -73,4 +79,53 @@ class ScientificValueTest {
         val decoded = Json.decodeFromString(ValueContainer.serializer(), json)
         assertEquals(container, decoded)
     }
+
+    @Test
+    fun testSplit() {
+        val (foot, inch) = 15(Inch).split(Foot, Inch)
+        assertEquals(1(Foot), foot)
+        assertEquals(3(Inch), inch)
+
+        val (fullInch, zeroFoot) = 15(Inch).split(Inch, Foot)
+        assertEquals(15(Inch), fullInch)
+        assertEquals(0(Foot), zeroFoot)
+
+        val (roundedFoot, roundedInch) = 11.999999999999(Inch).split(Foot, Inch)
+        assertEquals(1(Foot), roundedFoot)
+        assertEquals(0(Inch), roundedInch)
+
+        val (meter, centimeter) = 2.34(Meter).split(Centimeter, 1U)
+        assertEquals(2.3(Meter), meter)
+        assertEquals(4(Centimeter), centimeter)
+    }
+
+    @Test
+    fun testToComponents() {
```

Comment:
I'd test some odd cases like `1(Meter).toComponents(Kilometer, Kilometer)`

Reply:
Added a few. Better now or are you still missing a case?

Reply:
Now it's better, thanks!